src/bindings/python/src/openvino/utils.py

```diff
@@ -28,3 +29,27 @@ def add_openvino_libs_to_path() -> None:
                 # On Windows, with Python >= 3.8, DLLs are no longer imported from the PATH.
                 if (3, 8) <= sys.version_info:
                     os.add_dll_directory(os.path.abspath(lib_path))
+
+
+def deprecated(version: str = "", message: str = "") -> Callable[..., Any]:
+    """Prints deprecation warning and runs the function.
+
+    :param version: The version in which the code will be removed.
+    :param message: A message explaining why the function is deprecated and/or what to use instead.
+    """
+    def decorator(wrapped: Callable[..., Any]) -> Callable[..., Any]:
+        @wraps(wrapped)
+        def wrapper(*args: Any, **kwargs: Any) -> Callable[..., Any]:
+            from openvino.pyopenvino.util import deprecation_warning
+            deprecation_warning(wrapped.__name__, version, message)
+            return wrapped(*args, **kwargs)
+        return wrapper
+
+    # decorated function is assigned to version if there are no arguments passed
+    arg1 = version
+    if callable(arg1):
```

Comment:
Can we change `arg1` to something more descriptive? Also, type hint in L40 does not `version` could be `Callable`, just a string. I think I am missing something, why should it be checked if `version` is callable?

Did you mean that the first argument can be either a Python function object or a version, and branch `deprecated()` based on that? Do we expect `version` to be a function object?

Reply:
This is quite complicated. If you add the decorator with arguments (`@deprecated(message="msg", version="2.3")` ) version is a string, but if you add the decorator with no arguments (`@deprecated`) the version is the function it decorates

Reply:
We decided to simplify the implementation